server/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -609,6 +610,21 @@ public static SimpleQueryStringBuilder fromXContent(XContentParser parser) throw
         return qb;
     }
 
+    @Override
+    protected QueryBuilder doCoordinatorRewrite(CoordinatorRewriteContext coordinatorRewriteContext) {
+        if (fieldsAndWeights.containsKey(DataTierFieldMapper.NAME)) {
+            return tierFieldTermQueryCoordinatorRewriteIfPresent(
+                this,
+                true,
+                DataTierFieldMapper.NAME,
+                queryText,
+                coordinatorRewriteContext
+            );
+        } else {
+            return this;
+        }
+    }
+
```

Comment:
similar to query_string, I am not too sure that we should do this for simple query string